src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -1341,15 +1346,41 @@ def setup_regular_build(revision,
         environment.is_kernel_fuzzer_job()):
     build_class = CuttlefishKernelBuild
 
+  result = None
   build = build_class(
       base_build_dir,
       revision,
       build_url,
       target_weights=target_weights,
       build_prefix=build_prefix)
   if build.setup():
-    return build
-  return None
+    result = build
+  else:
+    return None
+
+  # Additional binaries to pull (for fuzzing engines such as Centipede).
+  extra_bucket_path = get_bucket_path('EXTRA_BUILD_BUCKET_PATH')
+  if extra_bucket_path:
+    # Import here as this path is not available in App Engine context.
+    from clusterfuzz._internal.bot.fuzzers import utils as fuzzer_utils
+    extra_build_urls = get_build_urls_list(extra_bucket_path)
+    extra_build_url = revisions.find_build_url(extra_bucket_path,
```

Comment:
I'm a bit unsure how something like this will work for chrome (and I assume google3).
In chrome different sanitizer builds are done by different builders that do not communicate, and there may not be a build for each sanitizer for a specific commit . How are builds from the same commit supposed to have the same revision?

Reply:
We'll need to get them to support it. I think the disk space benefit of our approach here is too much to pass up. 

The disk space benefit here is given two builds:

ASan: 5GB
NoSanitizer: 4GB

Rather than consuming 5GB+4GB to download a single zip and then unpacking, this can consume 5GB at most for the zips.

1. Download ASan (5GB)
2. Unpack (very minimal disk usage from selective unpack). The zip is deleted. 
3. Download NoSanitizer build (4GB.)
4. Unpack (again, very minimal disk usage from selective unpack). The zip is deleted.

Reply:
Also note that this implementation doesn't stop someone from having a single big zip. They just need to follow our expected directory format, and it'll still work.